jsonrpc/filter_manager.go

```diff
@@ -342,6 +349,116 @@ func (f *FilterManager) Exists(id string) bool {
 	return ok
 }
 
+// GetLogsForQuery return array of logs for given query
+func (f *FilterManager) GetLogsForQuery(query *LogQuery) ([]*Log, error) {
+	result := make([]*Log, 0)
+	parseReceipts := func(block *types.Block) error {
+		receipts, err := f.store.GetReceiptsByHash(block.Header.Hash)
+		if err != nil {
+			return err
+		}
+
+		for indx, receipt := range receipts {
+			for logIndx, log := range receipt.Logs {
+				if query.Match(log) {
+					result = append(result, &Log{
+						Address:     log.Address,
+						Topics:      log.Topics,
+						Data:        argBytes(log.Data),
+						BlockNumber: argUint64(block.Header.Number),
+						BlockHash:   block.Header.Hash,
+						TxHash:      block.Transactions[indx].Hash,
+						TxIndex:     argUint64(indx),
+						LogIndex:    argUint64(logIndx),
+					})
+				}
+			}
+		}
+
+		return nil
+	}
+
+	if query.BlockHash != nil {
+		block, ok := f.store.GetBlockByHash(*query.BlockHash, true)
+		if !ok {
+			return nil, fmt.Errorf("not found")
```

Comment:
Why is this an `Errorf` and not a standardized package error?

Reply:
Fixed dd2769c0533c4517d248f746a48bc10cf6ecc2b0